### DISCUSSION TITLE
How to batch composite hundreds of image

### DISCUSSION THREAD

**Discussion by AndehX** [ImageMagick/ImageMagick#6076] [Help]
### ImageMagick version

7.1.0-61

### Operating system, version and so on

Windows 11

### Description

Hi, I'd like to know if the following is possible.

I have a bunch character sprites for a game, and a corresponding bunch of sprites for the characters shadow.
I'd like to know if it's possible to merge each sprite with it's corresponding shadow?

The sprites are named 000-p.png all the way up to 271-p.png
The shadows are names 000-s.png all the way up to 271-s.png.

Each sprite needs to be combined with it's corresponding shadow. So 000-p.png combined with 000-s.png and so on...

Is this possible to automate this?

**Comment by fmw42** [comment]
How do they need to be combined?  Can you post an example?  To do this, you might be able to do it with mogrify. See https://imagemagick.org/Usage/basics/#mogrify_compose.   If that is not appropriate, then you would need to write a .bat script.

**Comment by snibgo** [comment]
> Each sprite needs to be combined with it's corresponding shadow.

If you show examples, perhaps we can show you the IM command to do that.

To do all the images, you probably need a shell `for` loop. You might loop through all the files named `???-p.png`. For each one found, change `-p.png` to `-s.png` for the other filename, then execute the IM command.
